akka-http-core/src/main/resources/reference.conf

```diff
@@ -107,6 +107,50 @@ akka.http {
     parsing = ${akka.http.parsing}
   }
 
+  host-connection-pool {
+    # The maximum number of parallel connections that a connection pool to a
+    # single host endpoint is allowed to establish. Must be greater than zero.
+    max-connections = 4
+
+    # The maximum number of times failed requests are attempted again,
+    # (if the request can be safely retried) before giving up and returning an error.
+    max-retries = 5
+
+    # The maximum number of open requests accepted into the pool across all
+    # materializations of any of its client flows.
+    # Protects against (accidentally) overloading a single pool with too many client flow materializations.
+    # Note that with N concurrent materializations the max number of open request in the pool
+    # will never exceed N * max-connections * pipelining-limit.
+    # Must be a power of 2 and > 0!
+    max-open-requests = 32
+
+    # The maximum number of requests that are dispatched to the target host in
+    # batch-mode across a single connection (HTTP pipelining).
+    # A setting of 1 disables HTTP pipelining, since only one request per
+    # connection can be "in flight" at any time.
+    # Set to higher values to enable HTTP pipelining.
+    # This value must be > 0.
+    # (Note that, independently of this setting, pipelining will never be done
+    # on a connection that still has a non-idempotent request in flight.
+    # See http://tools.ietf.org/html/rfc7230#section-6.3.2 for more info.)
+    pipelining-limit = 1
+
+    # The time after which an idle connection pool (without open connections)
+    # will automatically terminate itself. Set to `infinite` to disable idle timeouts.
+    idle-timeout = 30 s
+
+    # Grace period between beginning a connection pool shutdown and the (internal)
+    # gateway actor not accepting any more requests into the pool. Protects racy
+    # requests that were dispatched to the pool after the Shutdown message was
+    # queued.
+    # Set to zero to completely skip the grace-period (and accept the risk that
+    # requests sent around the time of an idle shutdown might get lost).
```

Comment:
> will it Just Work if I first send requests and then shut down the pool?

Yes, this is not a problem. The grace period is needed because we automatically shut down the pool after an idle-timeout. If I sent a request shortly after the idle-shutdown command has been queued this request would be lost without a grace period.